### DISCUSSION TITLE
Masking custom headers in logs

### DISCUSSION THREAD

**Discussion by Odenius** [open-policy-agent/community#370] [OPA and Rego] [answered]
Hi all, 

I'm wondering if there is a way to mask or upsert values from custom headers, the documentation "only" points to input, result and nd_builtin_cache. 

Our use case is that we need to utilize custom heardes for authorization towards a tunnel. We would therefore like to not have the secret and username stored in the logs if possible. 

I think the question was brought up a couple of years ago in https://github.com/open-policy-agent/opa/issues/3950.
Has anything changed since then regarding this topic?

**Comment by srenatus** [comment, marked as answer]
So I take it your headers make it to OPA's REST API, `/v1/data`, *after* having passed your tunnel, and are then logged with _debug_ log level? 

There's no masking for logs in OPA. The general guidance would be to not use debug logging in production settings. Masking only applies to decision logs, and they typically don't include any request headers. (They are also what you should rely on for auditability, not the debug request logs, I would thin.)

**Reply by Odenius** [reply]
Thanks for the answer.
